File: modulo1_C/exercicios/credit.c
#include<cs50.h>
#include<stdio.h>
#include<math.h>


long cartaoDeCredito();
int main(void){


long alvo=cartaoDeCredito();

//4003600000000014
//1234567891234567


long count=0;
long count1=0;
long count2=0;
long numeroDeDigito=0;
long valor=alvo;
long valor1=alvo;
long valor2=alvo;
int j=0;
int endNumber=0;


//Numero de digitos.
while(valor>1){
numeroDeDigito=numeroDeDigito+1;
int b=valor%10;
if(numeroDeDigito==16){
    endNumber=b;
}else if(numeroDeDigito==13){
    endNumber=b;
}
valor=valor/10;

}

//Primeira Operacao.
while(valor1>0){
int b =valor1%100;
int o=b/10;
int d=o*2;
 if(b>9){
    int f=d/10;
    int p=d%10;
    int g=f+p;
    count1=count1+g;
    }else{
        count1=count1+d;
    }
valor1=valor1/100;
}
//Guardando primeiro resultado na variavel count(resultado final).
count=count+count1;

//Segunda operacao.
while(valor2>0){
int b =valor2%100;
int i=b%10;
count2=count2+i;
valor2=valor2/100;
}

//Guardando segundo resultado na variavel count(resultado final).
count=count+count2;


//Apartir desse ponto ja temos o resultado que sera testado,pois ele e a soma de ambos resultados das operacoes.



if(count%10>0){
    printf("INVALID\n");
}else if(numeroDeDigito==15){
        printf("AMERICAN EXPRESS\n");
}else if(endNumber==4){
   printf("VISA\n");
}else if(numeroDeDigito==16){
   printf("MASTERCARD\n");
}






}
long cartaoDeCredito(){


    long numero=get_long("Numero do cartão: ");
    return numero;
}

File: modulo6_python/exercicios/credit.py
import math
import cs50


# 4003600000000014
# 1234567891234567
def main():



   input=cs50.get_int("Qual o numero do seu cartão? ")
  
   value=input
   count=0
   count1=0
   count2=0
   numeroDeDigito=0
   valor=value
   valor1=value
   valor2=value
   j=0
   endNumber=0


   while valor>1:
      numeroDeDigito=numeroDeDigito+1;
      b=valor%10;
      if numeroDeDigito==16:
        endNumber=math.floor(b)
      elif numeroDeDigito==13:
        endNumber=math.floor(b)
      valor=valor/10

   while valor1>0:
      b =valor1%100
      o=math.floor(b/10)
      d=o*2
      if b>9:
        f=math.floor(d/10)
        p=d%10
        g=f+p
        count1=math.floor(count1+g)
      else:
        count1=math.floor(count1+d)
      valor1=valor1/100

   count+=math.floor(count1)

   while valor2>0:
       b =valor2%100
       i=b%10
       count2+=i
       valor2=math.floor(valor2/100);

   count+=math.floor(count2)
   
   
   if count%10>0:
      print("INVALID\n")
   elif(numeroDeDigito==15):
      print("AMERICAN EXPRESS\n")
   elif(endNumber==4):
      print("VISA\n")
   elif numeroDeDigito==16:
      print("MASTERCARD")
   

main()